lib/runtime/router.js

```diff
@@ -77,24 +77,25 @@ export default class Router {
    * @private
    */
   handle(req, response) {
-    let request = new Request(req);
-    let routes = this.routes[request.method];
-    for (let i = 0; i < routes.length; i += 1) {
-      request.params = routes[i].match(request.path);
-      if (request.params) {
-        request.route = routes[i];
-        break;
-      }
-    }
-    if (!request.route) {
-      return this.handleError(request, response, new Errors.NotFound('Route not recognized'));
-    }
+    this.middleware.run(req, response, (error) => {
```

Comment:
I'm currently passing in the raw `req` object instead of the Denali `new Request(req)` object.  But some of my bugs apparently have nothing to do with the Request proxy here, so I'm happy to take that out if we think it'd be better ...